beacon-chain/state/v1/state_trie.go

```diff
@@ -509,13 +509,16 @@ func (b *BeaconState) rootSelector(ctx context.Context, field types.FieldIndex)
 
 func (b *BeaconState) recomputeFieldTrie(index types.FieldIndex, elements interface{}) ([32]byte, error) {
 	fTrie := b.stateFieldLeaves[index]
+	// We can't use trie trie's mutex because the trie's variable gets reassigned,
+	// and therefore we would call Unlock() on a different object.
+	fTrieMutex := b.stateFieldLeaves[index].RWMutex
```

Comment:
lets reuse the above var here:
```suggestion
	fTrieMutex := fTrie.RWMutex
```